contracts/test/ClaimTest.t.sol

```diff
@@ -0,0 +1,571 @@
+    uint256 constant USD = 10 ** 6;   // USDC precision
+
+    // Mainnet State Constants 
+    // Block 13499527 - Wednesday, October 27, 2021 12:58:18 PM UTC
+    // Using Orthogonal Pool for testing
+    uint256 constant PRINCIPAL_OUT     = 132_000_000_000000;
+    uint256 constant INTEREST_SUM      =     868_794_717158;
+    uint256 constant LL_USDC_BAL       =   6_516_420_406721;
+    uint256 constant SL_USDC_BAL       =     179_170_813216;
+    uint256 constant PD_USDC_BAL       =     122_108_154489;
+    uint256 constant TREASURY_USDC_BAL =     769_625_000000;
+
+    uint256 start;
+
+    // Mainnet State Variables
+    uint256 bpt_stakeLockerBal;
+    uint256 pool_principalOut;
+    uint256 pool_interestSum;
+    uint256 usdc_liquidityLockerBal;
+    uint256 usdc_stakeLockerBal;
+    uint256 usdc_poolDelegateBal;
+    uint256 usdc_treasuryBal;
+
+    Borrower borrower;
+
+    DebtLocker            debtLockerImplementation;
+    DebtLockerFactory     debtLockerFactory;
+    DebtLockerInitializer debtLockerInitializer;
+
+    IMapleLoan loanV2;
+
+    IMapleGlobalsLike globals = IMapleGlobalsLike(MAPLE_GLOBALS);
+    IPoolLike         pool    = IPoolLike(ORTHOGONAL_POOL);        // Using deployed Orthogonal Pool
+
+    IERC20 bpt  = IERC20(BALANCER_POOL);
+    IERC20 usdc = IERC20(USDC);
+    IERC20 wbtc = IERC20(WBTC);
+
+    MapleLoan            loanImplementation;
+    MapleLoanFactory     loanFactory;
+    MapleLoanInitializer loanInitializer;
+
+    function _createLoan(uint256 principalRequested, uint256 endingPrincipal) internal {
+        address[2] memory assets = [WBTC, USDC];
+
+        uint256[3] memory termDetails = [
+            uint256(10 days),  // 10 day grace period
+            uint256(30 days),  // 30 day payment interval
+            uint256(3)
+        ];
+
+        // 5 BTC @ ~$58k = $290k = 29% collateralized, interest only
+        uint256[3] memory requests = [uint256(5 * BTC), principalRequested, endingPrincipal];  
+
+        uint256[4] memory rates = [uint256(0.12e18), uint256(0), uint256(0.05e18), uint256(0.6e18)]; 
+
+        bytes memory arguments = loanInitializer.encodeArguments(address(borrower), assets, termDetails, requests, rates);
+
+        bytes32 salt = keccak256(abi.encodePacked("salt"));
+
+        loanV2 = IMapleLoan(borrower.mapleProxyFactory_createInstance(address(loanFactory), arguments, salt));
+    }
+
+    function _fundLoan() internal {
+        uint256 fundAmount       = 1_000_000 * USD;
+        uint256 establishmentFee = fundAmount * 25 * 90 / 365 / 10_000;  // Investor fee and treasury fee are both 25bps
+
+        pool_principalOut       = pool.principalOut();
+        pool_interestSum        = pool.interestSum();
+        usdc_liquidityLockerBal = usdc.balanceOf(ORTHOGONAL_LL);
+        usdc_stakeLockerBal     = usdc.balanceOf(ORTHOGONAL_SL);
+        usdc_poolDelegateBal    = usdc.balanceOf(ORTHOGONAL_PD);
+        usdc_treasuryBal        = usdc.balanceOf(MAPLE_TREASURY);
```

Comment:
We can actually delete this and move it to be below the fundLoan, we only need the state after the fundLoan occurs